src/fundus/scraping/html.py

```diff
@@ -274,99 +37,188 @@ class HTML:
     responded_url: str
     content: str
     crawl_date: datetime
-    source: "HTMLSource"
+    source: "SourceInfo"
 
 
 @dataclass(frozen=True)
-class HTMLSource:
+class SourceInfo:
     publisher: str
 
 
 @dataclass(frozen=True)
-class WarcSource(HTMLSource):
+class WarcSourceInfo(SourceInfo):
     warc_path: str
     warc_headers: Dict[str, str]
     http_headers: Dict[str, str]
 
 
 @dataclass(frozen=True)
-class WebSource(HTMLSource):
+class WebSourceInfo(SourceInfo):
     type: str
     url: str
 
 
-class FundusSource:
+class HTMLSource(Protocol):
+    @abstractmethod
+    def fetch(self, url_filter: Optional[URLFilter] = None) -> Iterator[HTML]:
+        ...
```

Comment:
This looks a bit weird to me: Is this class indented to be a protocol or an abstract class with an abstract method?

Reply:
The `abstractmethod` is a leftover.